alf/drivers/off_policy_driver.py

```diff
@@ -195,32 +201,26 @@ def train(self,
         experience = self._algorithm.preprocess_experience(experience)
 
         length = experience.step_type.shape[1]
-        if mini_batch_length is None:
-            mini_batch_length = length
-        else:
-            mini_batch_length = min(mini_batch_length, length)
+        mini_batch_length = (mini_batch_length or length)
+        assert length % mini_batch_length == 0, (
+            "length=%s mini_batch_length=%s" % (length, mini_batch_length))
 
         experience = tf.nest.map_structure(
             lambda x: tf.reshape(x, [-1, mini_batch_length] + list(x.shape[2:])
                                  ), experience)
 
         batch_size = experience.step_type.shape[0]
-        if mini_batch_size is None:
-            mini_batch_size = batch_size
-        else:
-            mini_batch_size = min(mini_batch_size, batch_size)
-
-        assert length % mini_batch_length == 0
+        mini_batch_size = (mini_batch_size or batch_size)
+        # The reason of this constraint is at L233
```

Comment:
be a bit more informative, e.g. reason is tf.reshape at L241

Reply:
updated